tests/o2a_libs/test_el_parser.py

```diff
@@ -42,36 +50,40 @@ class TestElParser(unittest.TestCase):
+            ('${header["host"]}', """{{params['header']["host"]}}"""),
+            ("${departments[deptName]}", "{{params['departments'][params['deptName']]}}"),
             (
                 "${requestScope['javax.servlet.forward.servlet_path']}",
-                "{{request_scope['javax.servlet.forward.servlet_path']}}",
+                "{{params['requestScope']['javax.servlet.forward.servlet_path']}}",
             ),
-            ("#{customer.lName}", "{{customer.l_name}}"),
-            ("#{customer.calcTotal}", "{{customer.calc_total}}"),
-            ('${wf:conf("jump.to") eq "ssh"}', '{{wf_conf("jump.to") == "ssh"}}'),
-            ('${wf:conf("jump.to") eq "parallel"}', '{{wf_conf("jump.to") == "parallel"}}'),
-            ('${wf:conf("jump.to") eq "single"}', '{{wf_conf("jump.to") == "single"}}'),
+            ("#{customer.lName}", "{{params['customer'].lName}}"),
+            ("#{customer.calcTotal}", "{{params['customer'].calcTotal}}"),
+            ('${wf:conf("jump.to") eq "ssh"}', '{{params["jump.to"] == "ssh"}}'),
+            ('${wf:conf("jump.to") eq "parallel"}', '{{params["jump.to"] == "parallel"}}'),
+            ('${wf:conf("jump.to") eq "single"}', '{{params["jump.to"] == "single"}}'),
             (
-                '${"bool" == bool ? print("ok") : print("not ok")}',
-                '{{print("ok") if "bool" == bool else print("not ok")}}',
+                '${"bool" == "bool" ? print("ok") : print("not ok")}',
+                '{{print("ok") if "bool" == "bool" else print("not ok")}}',
             ),
-            ('${f(x) ? print("ok") : print("not ok")}', '{{print("ok") if f(x) else print("not ok")}}'),
+            ('${f(2) ? print("ok") : print("not ok")}', '{{print("ok") if f(2) else print("not ok")}}'),
             ('${!false ? print("ok") : print("not ok")}', '{{print("ok") if !False else print("not ok")}}'),
             ("some pure text ${coord:user()}", "some pure text {{coord_user()}}"),
             (
                 "${nameNode}/user/${wf:user()}/${examplesRoot}/output-data/${outputDir}",
-                "{{name_node}}/user/{{wf_user()}}/{{examples_root}}/output-data/{{output_dir}}",
+                "{{params['nameNode']}}/user/{{params.props.merged.user.name}}"
+                "/{{params['examplesRoot']}}/output-data/{{params['outputDir']}}",
+            ),
+            (
+                "${YEAR}/${MONTH}/${DAY}/${HOUR}",
+                "{{params['YEAR']}}/{{params['MONTH']}}/{{params['DAY']}}/{{params['HOUR']}}",
             ),
-            ("${YEAR}/${MONTH}/${DAY}/${HOUR}", "{{year}}/{{month}}/{{day}}/{{hour}}"),
             ("pure text without any.function wf:function()", "pure text without any.function wf:function()"),
             (
                 "${fs:fileSize('/usr/foo/myinputdir') gt 10 * GB}",
-                "{{fs_file_size('/usr/foo/myinputdir') > 10 * gb}}",
+                "{{fs_file_size('/usr/foo/myinputdir') > 10 * 1073741824}}",
```

Comment:
Why not 1024 ** 3  here ? Is it somehow evaluated ?

Reply:
Yes when parsing it was evaluated. I changed it `1024 ** 3` to `"1024 ** 3"` so it will be more readable in template.